src/matcha_ml/core/core.py

```diff
@@ -90,3 +97,79 @@ def remove_state_lock() -> None:
     """Unlock remote state."""
     remote_state = RemoteStateManager()
     remote_state.unlock()
+
+
+@track(event_name=AnalyticsEvent.PROVISION)
+def provision(
+    location: str,
+    prefix: str,
+    password: str,
+    verbose: Optional[bool] = False,
+) -> MatchaState:
+    """Provision cloud resources using templates.
+
+    Args:
+        location (str): Azure location in which all resources will be provisioned.
+        prefix (str): Prefix used for all resources.
+        password (str): Password for ZenServer.
+        verbose (bool optional): additional output is show when True. Defaults to False.
+
+    Returns:
+        MatchaState: the information of the provisioned resources.
+
+    Raises:
+        MatchaError: If resources are already provisioned.
+        MatchaError: If prefix is not valid.
+        MatchaError: If region is not valid.
+    """
+    remote_state_manager = RemoteStateManager()
+
+    if remote_state_manager.is_state_stale():
+        # raise MatchaError("Matcha has detected a stale state file. This means that your local configuration is out of sync with the remote state, the resource group may have been removed.")
+        print_status(
+            build_warning_status(
+                "Matcha has detected a stale state file. This means that your local configuration is out of sync with the remote state, the resource group may have been removed. Deleting existing state config."
+            )
+        )
+        remote_state_manager.remove_matcha_config()
+        # Re-initialise remote state manager with empty state file
+        remote_state_manager = RemoteStateManager()
```

Comment:
TODO: possibly remove. Not sure what the best thing to do here is.

Reply:
Yeh, I would remove this. Has the added benefit of removing IO from here too